File: app/src/main/java/savexport/DatabaseController.java
package savexport;

import android.content.ContentValues;
import android.content.Context;
import android.database.sqlite.SQLiteDatabase;

import model.Game;

public class DatabaseController {
    private final MemoriaDbHelper memoriaDbHelper;

    public DatabaseController(Context paramContext) {
        this.memoriaDbHelper = new MemoriaDbHelper(paramContext);
    }

    public String insertData(Game game) {
        SQLiteDatabase db = this.memoriaDbHelper.getWritableDatabase();
        ContentValues contentValues = new ContentValues();
        contentValues.put(GameEntry.START_TIME, game.getInicio());
        contentValues.put(GameEntry.GAME_TIME, game.getDuracao());
        contentValues.put(GameEntry.HITS, game.getAcertos());
        contentValues.put(GameEntry.TIPS, game.getDicas());
        contentValues.put(GameEntry.MISSES, game.getErros());
        long l = db.insert(GameEntry.TABLE_NAME, null, contentValues);
        db.close();
        return (l == -1L) ? "Erro ao inserir registro" : "Registro inserido com sucesso";
    }
}

File: app/src/main/kotlin/savexport/DatabaseController.kt
package savexport

import android.content.ContentValues
import android.content.Context
import model.Game

class DatabaseController(paramContext: Context?) {
    private val memoriaDbHelper: MemoriaDbHelper
    fun insertData(game: Game): String {
        val db = memoriaDbHelper.writableDatabase
        val contentValues = ContentValues().apply {
            put(GameEntry.START_TIME, game.inicio)
            put(GameEntry.GAME_TIME, game.duracao)
            put(GameEntry.HITS, game.acertos)
            put(GameEntry.TIPS, game.dicas)
            put(GameEntry.MISSES, game.erros)
        }
        val result = db.insert(GameEntry.TABLE_NAME, null, contentValues)
        db.close()
        return if (result == -1L) "Erro ao inserir registro" else "Registro inserido com sucesso"
    }

    init {
        memoriaDbHelper = MemoriaDbHelper(paramContext)
    }
}